### PR TITLE
Add negation on needsForecast else case

### PR DESCRIPTION
**Pull request by jiskin**
Quantlib source gives:
Real f = timeSeries()[latestNeededDate];
return (f == Null<Real>());

which will return "true" when the latestNeededDate isn't found

### CODE REVIEW

**Comment by amaggiulli** [comment]
Error Message:
   Assert.Fail failed. ZC index does not forecast correctly for date 7/2/2007 from base date 6/1/2007 with fixing 207.3, correct:  207.799484810232, fix: 206.1, t 0.0833333333333333
Stack Trace:
   at TestSuite.QAssert.Fail(String message) in c:\projects\qlnet_develop\tests\QLNet.Tests\Utilities.cs:line 194
   at TestSuite.T_Inflation.testZeroTermStructure() in c:\projects\qlnet_develop\tests\QLNet.Tests\T_Inflation.cs:line 314

**Comment by jiskin** [comment]
hello,

is the test correct as I just followed the Quantlib logic...?

**Comment by amaggiulli** [comment]
I'm testing too , seems something wrong on the test schedules

File: src/QLNet/Indexes/InflationIndex.cs [hunk 1]
```diff
@@ -252,7 +252,7 @@ private bool needsForecast( Date fixingDate )
             // we're not sure, but the fixing might be there so we
             // check.  Todo: check which fixings are not possible, to
             // avoid using fixings in the future
-            return IndexManager.instance().getHistory( name() ).value().ContainsKey( latestNeededDate );
+            return !( IndexManager.instance().getHistory( name() ).value().ContainsKey( latestNeededDate ) );
```
Comment (igitur) [inline src/QLNet/Indexes/InflationIndex.cs]: Why not `return !timeSeries().value().ContainsKey(latestNeededDate)` to be consistent with QuantLib?

**Comment by igitur** [comment]
Think I solved it.
In https://github.com/amaggiulli/QLNet/blob/develop/tests/QLNet.Tests/T_Inflation.cs#L155 : remove the extra 2 indices (check QuantLib test).
In https://github.com/amaggiulli/QLNet/blob/develop/tests/QLNet.Tests/T_Inflation.cs#L160 :
Use `fixData.Length` instead of `rpiSchedule.Count` (check QuantLib test)

File: src/QLNet/Indexes/InflationIndex.cs (hunk 1 above)
Comment (jiskin) [inline src/QLNet/Indexes/InflationIndex.cs]: I didn't check why IndexManager is used here, so I kept the original implementation

**Pull request closed without merging**